logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreFailure.kt

```diff
@@ -82,10 +84,14 @@ internal inline fun <T : Any> wrapApiRequest(networkCall: () -> NetworkResponse<
         is NetworkResponse.Error -> {
             kaliumLogger.e(result.kException.stackTraceToString())
             val exception = result.kException
-            if (exception is KaliumException.GenericError && exception.cause is IOException) {
-                Either.Left(NetworkFailure.NoNetworkConnection(exception))
+            if (exception.cause?.message?.lowercase()?.contains("socks") == true) {
```

Comment:
Why this loose: any message containing the string "socks" is a `SocketError`? Can't we rely on a more specific exception type?

Reply:
the problem here is when there is an error from the proxy it throw SocketException which is Java.net exception 
and can't detect that from here , otherwise we have to have corefailure for each target, and I didn't think it's necessary for this case
If you have better approach please let me know

Reply:
If the exception is platform specific I guess we should wrap it in our own exception. If you don't want do this now at least add a TODO.